src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -370,6 +370,68 @@ public static void ObjectTree()
             Assert.Equal(3, child.MyOverflow["MyIntMissingChild"].GetInt32());
         }
 
+        [Fact]
+        public static void DeserializeIntoObjectProperty()
+        {
+            ClassWithExtensionPropertyAsObject obj;
```

Comment:
We should add a test with more than 1 null `JsonExtensionData` dictionaries. And one where the class contains a null `JsonExtensionData` dictionary and a dictionary property that we actually want serialized (both as null and with a value).